apps/apps.humanatlas.io/src/app/components/server-selector/server-selector.component.ts

```diff
@@ -0,0 +1,32 @@
+import { Component, input, model } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+import { MatSelectModule } from '@angular/material/select';
+import { Server } from '../../interfaces/server.interface';
+
+/**
+ * Server Selector Component
+ *
+ * This component is used in the HRA-API Rapidoc integration page
+ * to select a API server from a list of available servers.
+ *
+ * This component will take in a list of server options and lets
+ * the user select one. The selected server is available from the
+ * corresponding output binding.
+ */
+@Component({
+  selector: 'hra-server-selector',
+  imports: [FormsModule, MatSelectModule],
+  templateUrl: './server-selector.component.html',
+  styleUrl: './server-selector.component.scss',
+})
+export class ServerSelectorComponent {
+  /**
+   * input binding to get the list of servers
+   */
+  readonly servers = input<Server[]>([]);
+
+  /**
+   * two way binding for the selected server
+   */
+  readonly selectedServer = model<Server>(this.servers()[0]);
```

Comment:
There is a subtle bug here. Calling `this.servers()` during initialization will always return the default value, i.e. `[]`. Then getting `[][0]` returns undefined but the type `Server` indicate that it is never undefined.
Fix this by marking both servers and selectedServer as required input/model (`input.required<Server[]>()` and `model.required<Server>()`)